x-pack/legacy/plugins/code/server/lsp/language_servers.ts

```diff
@@ -72,12 +72,13 @@ export const CTAGS: LanguageServerDefinition = {
 export const LanguageServers: LanguageServerDefinition[] = [TYPESCRIPT, JAVA, GO, CTAGS];
 export const LanguageServersDeveloping: LanguageServerDefinition[] = [];
 
-export function enabledLanguageServers(server: ServerFacade) {
-  const devMode: boolean = server.config().get('env.dev');
+export function enabledLanguageServers(serverOptions: ServerOptions) {
+  const devMode: boolean = serverOptions.devMode;
 
   function isEnabled(lang: LanguageServerDefinition, defaultEnabled: boolean) {
     const name = lang.name;
-    const enabled = server.config().get(`xpack.code.lsp.${name}.enabled`);
+    // @ts-ignore
+    const enabled = serverOptions.lsp[name] && serverOptions.lsp[name].enabled;
```

Comment:
`LspOptions` typings are not full. there is no info about available languages and typings are broken due to https://github.com/elastic/kibana/pull/46664/files#r329619654

Reply:
This is a legacy issue for the `LspOptions` specifically. I will defer to @spacedragon to fix this later. In short, I think let's focus more on the major NP service migration in this PR.